be/src/runtime/routine_load/routine_load_task_executor.cpp

```diff
@@ -227,6 +227,28 @@ void RoutineLoadTaskExecutor::exec_task(
     // commit txn
     HANDLE_ERROR(_exec_env->stream_load_executor()->commit_txn(ctx), "commit failed");
 
+    // commit kafka offset
+    switch (ctx->load_src_type) {
+        case TLoadSourceType::KAFKA: {
+            std::shared_ptr<DataConsumer> consumer;
+            HANDLE_ERROR(_data_consumer_pool.get_consumer(ctx, &consumer), "failed to get consumers");
```

Comment:
It's better not return any error if we failed to do something after txn is committed.